common/db/db.js

```diff
@@ -165,6 +165,47 @@ export default class Db {
         return this.getResultsForCurrentUser(expName);
     }
 
+    async getResultsForExperiment(experimentName) {
+        const results = [];
+        const baseParams = {
+            TableName: this.experimentTable,
+            FilterExpression: "begins_with(experimentDateTime, :experimentName)",
+            ExpressionAttributeValues: {":experimentName": experimentName},
+        };
+        let lastKey = null;
+        try {
+            do {
+                const response = await this.scan({
+                    ...baseParams,
+                    ...(lastKey === null ? {} : {ExclusiveStartKey: lastKey}),
+                });
+                response.Items.forEach(item => {
+                    const parts = item.experimentDateTime.split("|");
+                    if (parts.length != 3) {
+                        throw new Error(`Unexpected experimentDateTime value: ${i.experimentDateTime}. Expected three parts, but found ${parts.length}.`)
+                    }
+                    const [experiment, dateTime, _index] = parts;
+                    results.push({
+                        dateTime,
+                        experiment,
+                        identityId: item.identityId,
```

Comment:
Let's use userId here instead of identityId. If we ever need to link this back to a particular user record that will save us a step.